test/SILGen/moveonly_escaping_closure.swift

```diff
@@ -0,0 +1,1583 @@
+// CHECK:   apply {{%.*}}([[LOADED]])
+// CHECK:   destroy_value [[LOADED]]
+// CHECK:   end_access [[ACCESS]]
+//
+// CHECK:   [[ACCESS:%.*]] = begin_access [deinit] [unknown] [[PROJECT]]
+// CHECK:   [[CHECK:%.*]] = mark_must_check [assignable_but_not_consumable] [[ACCESS]]
+// CHECK:   [[LOADED:%.*]] = load [take] [[CHECK]]
+// CHECK:   apply {{%.*}}([[LOADED]])
+// CHECK:   end_access [[ACCESS]]
+//
+// CHECK:   [[ACCESS:%.*]] = begin_access [deinit] [unknown] [[PROJECT]]
+// CHECK:   [[CHECK:%.*]] = mark_must_check [assignable_but_not_consumable] [[ACCESS]]
+// CHECK:   [[LOADED:%.*]] = load [take] [[CHECK]]
+// CHECK:   apply {{%.*}}([[LOADED]])
+// CHECK:   end_access [[ACCESS]]
+//
+// CHECK:   [[READ_ACCESS:%.*]] = begin_access [read] [unknown] [[PROJECT]]
+// CHECK:   [[CHECK:%.*]] = mark_must_check [no_consume_or_assign] [[READ_ACCESS]]
+// CHECK:   [[LOADED_READ:%.*]] = load [copy] [[CHECK]]
+// CHECK:   [[DEINIT_ACCESS:%.*]] = begin_access [deinit] [unknown] [[PROJECT]]
+// CHECK:   [[CHECK:%.*]] = mark_must_check [assignable_but_not_consumable] [[DEINIT_ACCESS]]
+// CHECK:   [[LOADED_TAKE:%.*]] = load [take] [[CHECK]]
+// CHECK:   apply {{%.*}}([[LOADED_READ]], [[LOADED_TAKE]])
+// CHECK:   end_access [[DEINIT_ACCESS]]
+// CHECK:   destroy_value [[LOADED_READ]]
+// CHECK:   end_access [[READ_ACCESS]]
+// CHECK: } // end sil function '$s16moveonly_closure36testConsumingEscapeClosureCaptureVaryyyycnFyycfU_'
+func testConsumingEscapeClosureCaptureVar(_ f: consuming @escaping () -> ()) {
+    var x = SingleElt()
+    x = SingleElt()
+    f = {
+        borrowVal(x)
+        consumeVal(x) // expected-error {{'x' was consumed but it is illegal to consume a noncopyable mutable capture of an escaping closure. One can only read from it or assign over it}}
+        consumeVal(x) // expected-error {{'x' was consumed but it is illegal to consume a noncopyable mutable capture of an escaping closure. One can only read from it or assign over it}}
+        borrowConsumeVal(x, x) // expected-error {{'x' was consumed but it is illegal to consume a noncopyable mutable capture of an escaping closure. One can only read from it or assign over it}}
+        // expected-error @-1 {{overlapping accesses, but deinitialization requires exclusive access; consider copying to a local variable}}
+        // expected-note @-2 {{conflicting access is here}}
+    }
+    f()
+}
+
+/////////////////////////////
+// MARK: Let Capture Tests //
+/////////////////////////////
+
+// NOTE: In the let tests below, we often times have multiple mark_must_check on
+// the same project_box. If we just looked at the SILGen output, the move
+// checker would potentially not work correctly upon that code and would emit
+// weird diagnostics since the AccessVisitor will visit /all/ uses on all of the
+// mark_must_check when we process each individual mark must check. Luckily this
+// is not actually an issue since when allocbox to stack runs, we specialize the
```

Comment:
The var-capture tests have multiple `mark_must_check`'s. Is it OK for vars but not for lets? If so, maybe worth noting that too. Also, would this explainer make more sense elsewhere in the source code rather than a test, for discoverability?

Reply:
This comment actually needs to be eliminated since we do not use the AccessVisitor anymore.